pylti/flask.py

```diff
@@ -96,15 +96,32 @@ def verify(self):
 
     def _verify_any(self):
         """
-        Verify that request is in session or initial request
+        Verify that an initial request has been made, or failing that, that
+        the request is in the session
 
         :raises: LTIException
         """
         log.debug('verify_any enter')
-        try:
-            self._verify_session()
-        except LTINotInSessionException:
+
+        # Check to see if there is a new LTI launch request incoming
+        newrequest = False
+        if flask_request.method == 'POST':
+            params = flask_request.form.to_dict()
+            initiation = "basic-lti-launch-request"
+            if params.get("lti_message_type", None) == initiation:
+                newrequest = True
+                # Scrub the session of the old authentication
+                for prop in LTI_PROPERTY_LIST:
+                    if session.get(prop, None):
+                        del session[prop]
+                session[LTI_SESSION_KEY] = False
+
+        # Attempt the appropriate validation
+        # Both of these methods raise LTIException as necessary
+        if newrequest:
             self.verify_request()
+        else:
+            self._verify_session()
```

Comment:
i think it should be , because it can be initial request without `lti_message_type: basic-lti-launch-request`. We should keep in mind fall back option, where apps do not incorporate `lti_message_type: basic-lti-launch-request` prop in them

```python
try:
            self._verify_session()
        except LTINotInSessionException:
            self.verify_request()
```